File: app/src/main/java/com/example/pam_project/features/lists/edit/EditListView.java
package com.example.pam_project.features.lists.edit;

import com.example.pam_project.features.categories.list.CategoryInformation;
import com.example.pam_project.features.lists.list.ListInformation;

import java.util.List;

public interface EditListView {

    void bindCategories(final List<CategoryInformation> model);

    void bindList(final ListInformation model);

    void onListDelete();

    void onListDeletedError();

    void onListUpdatedError();

    void onCategoriesReceivedError();

    void showDeleteDialog();
}

File: app/src/main/java/com/example/pam_project/features/lists/edit/EditListView.kt
package com.example.pam_project.features.lists.edit

import com.example.pam_project.features.categories.list.CategoryInformation
import com.example.pam_project.features.lists.list.ListInformation

interface EditListView {
    fun bindCategories(model: List<CategoryInformation?>?)
    fun bindList(model: ListInformation?)
    fun onListDelete()
    fun onListDeletedError()
    fun onListUpdatedError()
    fun onCategoriesReceivedError()
    fun showDeleteDialog()
    fun onListReceivedError()
}